src/iOS.Core/Controllers/LoginPasswordViewController.cs

```diff
@@ -0,0 +1,218 @@
+
+            base.ViewDidLoad();
+        }
+
+        public override void ViewDidAppear(bool animated)
+        {
+            base.ViewDidAppear(animated);
+            if (string.IsNullOrWhiteSpace(EmailCell.TextField.Text))
+            {
+                EmailCell.TextField.BecomeFirstResponder();
+            }
+            else
+            {
+                MasterPasswordCell.TextField.BecomeFirstResponder();
+            }
+        }
+
+        protected async Task LogInAsync()
+        { 
+            var email = EmailCell.TextField.Text;
+            var password = MasterPasswordCell.TextField.Text;
+            if (Xamarin.Essentials.Connectivity.NetworkAccess == Xamarin.Essentials.NetworkAccess.None)
+            {
+                var connectivityAlert = Dialogs.CreateAlert(AppResources.InternetConnectionRequiredTitle,
+                    AppResources.InternetConnectionRequiredMessage, AppResources.Ok);
+                PresentViewController(connectivityAlert, true, null);
+                return;
+            }
+            if (string.IsNullOrWhiteSpace(email))
+            {
+                var emailEmptyAlert = Dialogs.CreateAlert(AppResources.AnErrorHasOccurred,
+                    string.Format(AppResources.ValidationFieldRequired, AppResources.EmailAddress),
+                    AppResources.Ok, (a) => EmailCell.TextField.BecomeFirstResponder());
+                PresentViewController(emailEmptyAlert, true, null);
+                return;
+            }
+            if (!email.Contains("@"))
+            {
+                var emailInvalidAlert = Dialogs.CreateAlert(AppResources.AnErrorHasOccurred,
+                    AppResources.InvalidEmail, AppResources.Ok,(a) => EmailCell.TextField.BecomeFirstResponder());
+                PresentViewController(emailInvalidAlert, true, null);
+                return;
+            }
+            if (string.IsNullOrWhiteSpace(password))
+            {
+                var passwordEmptyAlert = Dialogs.CreateAlert(AppResources.AnErrorHasOccurred,
+                    string.Format(AppResources.ValidationFieldRequired, AppResources.MasterPassword),
+                    AppResources.Ok,(a) => MasterPasswordCell.TextField.BecomeFirstResponder());
+                PresentViewController(passwordEmptyAlert, true, null);
+                return;
+            }
+            
+            try
+            {
+                await _deviceActionService.ShowLoadingAsync(AppResources.LoggingIn);
+                var response = await _authService.LogInAsync(email, password);
+                await _deviceActionService.HideLoadingAsync();
+                if (response.TwoFactor)
+                {
+                    // TODO Segue to Two Factor page
+                }
+                else
+                {
+                    var task = Task.Run(async () => await _syncService.FullSyncAsync(true));
```

Comment:
This method is already async, should just be reduced to:
`await _syncService.FullSyncAsync(true);`

Reply:
Updaed!